lib/session-store.js

```diff
@@ -0,0 +1,74 @@
+/*
+// https://github.com/expressjs/session#session-store-implementation
+//
+// Uses PL database as a express-session store
+*/
+const session = require('express-session');
+const _ = require('lodash');
+const util = require('util');
+
+const config = require('./config');
+const sqldb = require('@prairielearn/postgres');
+const sql = sqldb.loadSqlEquiv(__filename);
+
+class plSessionStore extends session.Store {
+  //
+  // Required
+  //
+  set = util.callbackify(async (sid, session) => {
+    let params = {
+      sid,
+      session: JSON.stringify(session),
+    };
+    await sqldb.queryOneRowAsync(sql.upsert, params);
+  });
+
+  get = util.callbackify(async (sid) => {
+    let params = {
+      sid,
+      expirationInSeconds: config.sessionStoreExpireSeconds,
```

Comment:
Let's pass the expiration in the constructor so it's very straightforward to eventually lift this into a module that doesn't directly depend on `./config`.